src/iop/useless.c

```diff
@@ -184,32 +249,22 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
     g_hash_table_replace(piece->raster_masks, GINT_TO_POINTER(mask_id), mask);
 }
 
-/** optional: if this exists, it will be called to init new defaults if a new image is loaded from film strip
- * mode. */
-void reload_defaults(dt_iop_module_t *module)
-{
-  // change default_enabled depending on type of image, or set new default_params even.
-
-  // if this callback exists, it has to write default_params and default_enabled.
-}
-
 /** init, cleanup, commit to pipeline */
 void init(dt_iop_module_t *module)
 {
-  // we don't need global data:
-  module->data = NULL; // malloc(sizeof(dt_iop_useless_global_data_t));
-  module->params = calloc(1, sizeof(dt_iop_useless_params_t));
-  module->default_params = calloc(1, sizeof(dt_iop_useless_params_t));
-  // our module is disabled by default
-  // by default:
-  module->default_enabled = 0;
-  module->params_size = sizeof(dt_iop_useless_params_t);
-  module->gui_data = NULL;
-  // init defaults:
-  dt_iop_useless_params_t tmp = (dt_iop_useless_params_t){ .checker_scale = 50, .factor = 0.5 };
-
-  memcpy(module->params, &tmp, sizeof(dt_iop_useless_params_t));
-  memcpy(module->default_params, &tmp, sizeof(dt_iop_useless_params_t));
+  // allocates memory for a module instance and fills default_params.
+  // if this callback is not provided, the standard implementation in
+  // dt_iop_default_init is used, which looks at the $DEFAULT introspection tags
+  // in the comments of the params_t struct definition.
+  // an explicit implementation of init is only required if not all fields are
+  // fully supported by dt_iop_default_init, for example arrays with non-identical values.
+  // in that case, dt_iop_default_init can be called first followed by additional initialisation.
+  // the values in params will not be used and default_params can be overwritten by
+  // reload_params on a per-picture basis.
+  dt_iop_default_init(module);
+
+  // our module is enabled by default:
+  module->default_enabled = 1;
```

Comment:
Might also be worth referring to `module->hide_enable_button` as well?

Reply:
could do, but this isn't intended to be exhaustive. just one non-default setting to justify not falling back completely to the default init (be not providing our own init at all).

Reply:
ok